collect_app/src/main/java/org/odk/collect/android/geo/GoogleMapFragment.java

```diff
@@ -160,11 +166,21 @@ public class GoogleMapFragment extends SupportMapFragment implements
     @Override public void onAttach(@NonNull Context context) {
         super.onAttach(context);
         DaggerUtils.getComponent(context).inject(this);
+
+        GoogleMapConfigurator configurator = new GoogleMapConfigurator(
+                KEY_GOOGLE_MAP_STYLE, R.string.basemap_source_google,
+                new GoogleMapTypeOption(GoogleMap.MAP_TYPE_NORMAL, R.string.streets),
+                new GoogleMapTypeOption(GoogleMap.MAP_TYPE_TERRAIN, R.string.terrain),
+                new GoogleMapTypeOption(GoogleMap.MAP_TYPE_HYBRID, R.string.hybrid),
+                new GoogleMapTypeOption(GoogleMap.MAP_TYPE_SATELLITE, R.string.satellite)
+        );
+
+        mapFragmentDelegate = new MapFragmentDelegate(configurator, settingsProvider.getUnprotectedSettings(), this::onConfigChanged);
     }
 
     @Override public void onStart() {
         super.onStart();
-        mapProvider.onMapFragmentStart(this);
```

Comment:
This is the code I really needed to be rid of. Delegating the config and listener registration to the singleton `MapProvider` requires it to have created all the `MapFragment` instances, so it can grab them out of the two maps (`listenersByMap` and `configuratorsByMap`) later. This means that recreated Fragments (instantiated during rotation for instance) would not work properly as they would never get registered with the `MapProvider`. As far as I can see, other than that being the way it works, there's no reason for any of that to be singleton state and the `MapFragment` instances should be able to handle their own configuration and listener registration. 

I've also added a `MapFragmentDelegate` to cut down any duplication that was added by pulling the code out of `MapProvider` and into the `MapFragment` implementations.